python/common/org/python/types/Range.java

```diff
@@ -118,4 +121,50 @@ public Range(org.python.Object start, org.python.Object stop, org.python.Object
         }
     }
 
+    @org.python.Method(
+        __doc__ = "Implement __len__(self)."
+    )
+    public org.python.types.Int __len__() {
```

Comment:
`__repr__` should be returning an Object, but IIRC `__iter__` needs to return an `org.python.Iterable` (the interface, not the class) to make iteration operations work.